### PR TITLE
Generate doc comments for parameters

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Add a blurb about optional, constant values.

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Fixes https://github.com/Azure/autorest.go/issues/726

File: src/generator/operations.ts [hunk 1]
```diff
@@ -128,17 +128,23 @@ export async function generateOperations(session: Session<CodeModel>): Promise<O
...
     }
 
     const methodParams = new Array<string>();
+    const paramDocs = new Array<string>();
     if (<boolean>session.model.language.go!.azureARM) {
       // AzureARM is the simplest case, no parametertized host etc
       imports.add('github.com/Azure/azure-sdk-for-go/sdk/azcore');
       emitClientParams();
       methodParams.push('credential azcore.TokenCredential');
+      paramDocs.push('// credential - the credential used to authenticate the request.');
```
Comment (chlowell) [inline src/generator/operations.ts:145]: Trying to fit in a getting started hint:
```suggestion
      paramDocs.push('// credential - used to authorize requests. Usually a credential from azidentity.');
```

File: test/compute/2019-12-01/armcompute/zz_generated_containerservices_client.go [hunk 2]
```diff
@@ -30,6 +30,9 @@ type ContainerServicesClient struct {
 }
 
 // NewContainerServicesClient creates a new instance of ContainerServicesClient with the specified values.
+// subscriptionID - Subscription credentials which uniquely identify Microsoft Azure subscription. The subscription ID forms part of the URI for every service call.
```
Comment (chlowell) [inline test/compute/2019-12-01/armcompute/zz_generated_containerservices_client.go:33]: Unfortunate overloading of "credentials". Needs a swagger edit to change?

File: test/maps/azalias/zz_generated_alias_client.go [hunk 3]
```diff
@@ -19,12 +19,17 @@ import (
...
 	apiVersion *string
-	pl         runtime.Pipeline
+	pl runtime.Pipeline
 }
 
 // newAliasClient creates a new instance of aliasClient with the specified values.
+// geography - This parameter specifies where the Azure Maps Creator resource is located.  Valid values are us and eu.
+// apiVersion - Api Version
+Specifying any value will set the value to 2.0.
+Specifying any value will set the value to 2.0.
```
Comment (chlowell) [inline test/maps/azalias/zz_generated_alias_client.go:31]: One reason the build breaks

File: test/compute/2019-12-01/armcompute/zz_generated_containerservices_client.go (hunk 2 above)
Comment (jhendrixMSFT) [reply test/compute/2019-12-01/armcompute/zz_generated_containerservices_client.go:33]: Unfortunately yes.

File: test/maps/azalias/zz_generated_alias_client.go (hunk 3 above)
Comment (jhendrixMSFT) [reply test/maps/azalias/zz_generated_alias_client.go:31]: fixed

File: src/generator/operations.ts (hunk 1 above)
Comment (jhendrixMSFT) [reply src/generator/operations.ts]: Good idea, fixed

**Review by chlowell — APPROVED** [review]

**Pull request merged**